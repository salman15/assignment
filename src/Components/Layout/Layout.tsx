import React from "react";
import styled from "styled-components";

interface DefaultLayoutProps {
  children: React.ReactNode;
}

export const Container: React.FC<DefaultLayoutProps> = ({ children }) => {
  return <ContainerLayout>{children}</ContainerLayout>;
};

const ContainerLayout = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  width: 100%;
  height: 100%;
  background-color: #ededed;
`;

export const WarningText: React.FC<DefaultLayoutProps> = ({ children }) => {
  return <WarningTextLayout>{children}</WarningTextLayout>;
};

const WarningTextLayout = styled.p`
  color: red;
`;

export const Column: React.FC<DefaultLayoutProps> = ({ children }) => {
  return <ColumnLayout>{children}</ColumnLayout>;
};

const ColumnLayout = styled.div`
  display: flex;
  flex-direction: column;
  padding: 32px;
  background-color: white;
  border-radius: 4px;
  box-shadow: 0 2.8px 2.2px rgba(0, 0, 0, 0.034),
    0 6.7px 5.3px rgba(0, 0, 0, 0.048), 0 12.5px 10px rgba(0, 0, 0, 0.06),
    0 22.3px 17.9px rgba(0, 0, 0, 0.072), 0 41.8px 33.4px rgba(0, 0, 0, 0.086),
    0 100px 80px rgba(0, 0, 0, 0.12);
`;
