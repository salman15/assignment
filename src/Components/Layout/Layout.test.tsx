import React from "react";
import { render } from "@testing-library/react";
import { Column, Container, WarningText } from "./Layout";

describe("Test the Container component", () => {
  it("Test if Container component renders with children", () => {
    // Act
    const { container, getByText } = render(
      <Container>
        <p>I'm a Container</p>
      </Container>
    );

    // Assert
    const pElement = getByText(/I'm a Container/i);
    expect(container).toMatchSnapshot();
    expect(pElement).toBeInTheDocument();
  });
});

describe("Test the WarningText component", () => {
  it("Test if WarningText component renders with children", () => {
    // Act
    const { container, getByText } = render(
      <WarningText>
        <p>I'm a WarningText</p>
      </WarningText>
    );

    // Assert
    const pElement = getByText(/I'm a WarningText/i);
    expect(container).toMatchSnapshot();
    expect(pElement).toBeInTheDocument();
  });
});

describe("Test the Column component", () => {
  it("Test if Column component renders with children", () => {
    // Act
    const { container, getByText } = render(
      <Column>
        <p>I'm a Column</p>
      </Column>
    );

    // Assert
    const pElement = getByText(/I'm a Column/i);
    expect(Column).toMatchSnapshot();
    expect(pElement).toBeInTheDocument();
  });
});
