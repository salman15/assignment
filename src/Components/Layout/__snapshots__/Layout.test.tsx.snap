// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Column component Test if Column component renders with children 1`] = `[Function]`;

exports[`Test the Container component Test if Container component renders with children 1`] = `
<div>
  <div
    class="sc-bdfBwQ dtXNoS"
  >
    <p>
      I'm a Container
    </p>
  </div>
</div>
`;

exports[`Test the WarningText component Test if WarningText component renders with children 1`] = `
<div>
  <p
    class="sc-gsTCUz wIMML"
  >
    <p>
      I'm a WarningText
    </p>
  </p>
</div>
`;
